p2p/swarm.go

```diff
@@ -627,25 +627,29 @@ func (s *swarm) startNeighborhood() error {
 }
 
 func (s *swarm) peersLoop() {
+	var moreLock uint32 = 0
 loop:
 	for {
 		select {
 		case <-s.morePeersReq:
 			s.lNode.Debug("loop: got morePeersReq")
-			go s.askForMorePeers()
+			if atomic.CompareAndSwapUint32(&moreLock, 0, 1) {
+				go s.askForMorePeers(&moreLock)
```

Comment:
all atomics can be refactored out by:
```
if s.idle{
go func() {
 s.askForMorePeers()
 s.idle = true
}

Reply:
I've decided to just remove locking and call `askForMorePeers` synchronously because our chan is buffered, we won't block when it sends on askForMorePeers, but it will never run more than one instance in parallel.